sdk/servicelinker/azure-resourcemanager-servicelinker/src/samples/java/com/azure/resourcemanager/servicelinker/CreateServiceLinker.java

```diff
@@ -0,0 +1,184 @@
+package com.azure.resourcemanager.servicelinker;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.Region;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.identity.AzureAuthorityHosts;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.appplatform.models.SkuName;
+import com.azure.resourcemanager.appplatform.models.SpringApp;
+import com.azure.resourcemanager.appplatform.models.SpringService;
+import com.azure.resourcemanager.appservice.models.PricingTier;
+import com.azure.resourcemanager.appservice.models.RuntimeStack;
+import com.azure.resourcemanager.appservice.models.WebApp;
+import com.azure.resourcemanager.keyvault.models.Vault;
+import com.azure.resourcemanager.msi.models.Identity;
+import com.azure.resourcemanager.servicelinker.models.AzureResource;
+import com.azure.resourcemanager.servicelinker.models.ClientType;
+import com.azure.resourcemanager.servicelinker.models.LinkerResource;
+import com.azure.resourcemanager.servicelinker.models.SecretAuthInfo;
+import com.azure.resourcemanager.servicelinker.models.SourceConfiguration;
+import com.azure.resourcemanager.servicelinker.models.UserAssignedIdentityAuthInfo;
+import com.azure.resourcemanager.servicelinker.models.ValueSecretInfo;
+import com.azure.resourcemanager.sql.models.SqlDatabase;
+import com.azure.resourcemanager.sql.models.SqlServer;
+import reactor.core.publisher.Mono;
+
+import java.util.Locale;
+import java.util.UUID;
+
+public class CreateServiceLinker {
+    private static final String USER_TOKEN_HEADER = "x-ms-serviceconnector-user-token";
+
+    /**
+     * Main entry point.
+     *
+     * @param args the parameters
+     */
+    public static void main(String[] args) throws Exception {
+        // Authentication environment variables need to be set: AZURE_CLIENT_ID,AZURE_TENANT_ID,AZURE_CLIENT_SECRET,AZURE_SUBSCRIPTION_ID
+        TokenCredential credential = new DefaultAzureCredentialBuilder()
+            .authorityHost(AzureAuthorityHosts.AZURE_PUBLIC_CLOUD)
+            .build();
+
+        AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        // create a policy for setting user token into a specific header for some special scenarios
```

Comment:
will there be readme.md file? pls help to add more specific description on uasge of user token. keyvault, mysql flexible, network feature

Reply:
Does the "mysql flexible" need userToken permanently? If not, I'd not add it to readme as published version cannot be changed.

Readme would be cleaned every generation. I could add it but it is not a good choice to maintain here.

Reply:
readme under samples will not be cleared, and you need that to be published to sample browse https://docs.microsoft.com/en-us/samples/browse/

readme under root will be cleared, though we should be able to add some way to preserve part of the customized readme.

Reply:
Added a readme in sample folder https://github.com/Azure/azure-sdk-for-java/pull/28333/files#diff-ba0e6092eadc8fcf817568fec7a412df4b829a7a8dd0b7c28994c94126a384cc